[neoslippi] Add neoslippi

Kered13 opened the pull request:
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says.
- [X] The installed as the "copyright" file matches what upstream says.
- [X] The source code of the component installed comes from an authoritative source.
- [X] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

Kered13 committed:
Add neoslippi

jimwang118 commented:
I got the following error when testing the usage of neoslippi.
```
Severity	Code	Description	Project	File	Line	Suppression State	Details
Error		CMake Error at F:/neoslippi/scripts/buildsystems/vcpkg.cmake:859 (_find_package):
  Could not find a package configuration file provided by "NeoSlippi" with
  any of the following names:

    NeoSlippiConfig.cmake
    neoslippi-config.cmake

  Add the installation prefix of "NeoSlippi" to CMAKE_PREFIX_PATH or set
  "NeoSlippi_DIR" to a directory containing one of the above files.  If
  "NeoSlippi" provides a separate development package or SDK, be sure it has
  been installed.		F:/neoslippi/scripts/buildsystems/vcpkg.cmake	859		

```

Kered13 commented:
@jimwang118 Hmm, are you sure? I just tested it locally with a Visual Studio solution and it had no trouble picking it up. I also compared the port files to my previous project, `cppslippi`, and they are basically the same (except for name, SHA, etc. of course). Could you try building `cppslippi` and see if that one works for you? Could you also provide any other information, like error logs, that might help me investigate the problem?

jimwang118 commented:
> @jimwang118 Hmm, are you sure? I just tested it locally with a Visual Studio solution and it had no trouble building it. I also compared the port files to my previous project, `cppslippi`, and they are basically the same (except for name, SHA, etc. of course). Could you try building `cppslippi` and see if that one works for you? Could you also provide any other information, like error logs, that might help me investigate the problem?

I created a cmake project and called the post-installation prompt usage in cmakelists.txt, and then the error mentioned above occurred. I checked that the config file was not generated in the `installed/x-windows/share/neoslippi` directory.

dg0yt commented on ports/neoslippi/usage:
```diff
@@ -0,0 +1,4 @@
+The package neoslippi provides CMake targets:
+
+    find_package(NeoSlippi CONFIG REQUIRED)
+    target_link_libraries(main PRIVATE NeoSlippi::NeoSlippi)
```
```suggestion
neoslippi provides CMake targets:

  find_package(NeoSlippi CONFIG REQUIRED)
  target_link_libraries(main PRIVATE NeoSlippi::NeoSlippi)

```

dg0yt reviewed (commented):
CI file list:
~~~
neoslippi:x64-windows:/share/neoslippi/NeoSlippiTargets-debug.cmake
neoslippi:x64-windows:/share/neoslippi/NeoSlippiTargets-release.cmake
neoslippi:x64-windows:/share/neoslippi/NeoSlippiTargets.cmake
~~~
There really is no `NeoSlippiConfig.cmake`.

jimwang118 commented:
Usage test pass with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

Kered13 commented:
Ping on this PR. I think all the issues are fixed now?

data-queue commented:
We look for multiple downstream projects consuming each library before accepting it into the core catalog. Can you help us by linking multiple downstream projects authored by other individuals/projects?

Otherwise, this seems like a great candidate for self-publishing (port overlays or a custom registry).

Kered13 commented:
There are no downstream users because I am trying to publish the package. And I highly doubt there will be any downstream users (other than myself, but I did not write this for my own consumption) if it is not published. VCPKG is how I intend to make it available to downstream users.

I cannot find anything on the VCPKG documentation pages saying that projects should already have downstream users, and I did not receive any comments like this when I published my previous package, cppslippi (#30315).

jimwang118 commented:
If you confirm the repair is complete, click "ready for review".

data-queue commented:
Hi @Kered13, unfortunately we can't accept the port at this time. Any port accepted in the catalog is rebuilt when any of its dependencies update and a few times a week in CI for all time and we see this as a high ongoing maintenance cost. So, we are more selective in what we add to the catalog (the requirements I listed above). This is a new policy from the team, applying only to new ports. While we can't add it to the main catalog, you can make the library available through an overlay port or a custom registry.

Closing this PR.

vicroms commented:
Hi @Kered13

We apologize for closing this PR prematurely. 

As we mentioned in our previous comment, we are considering new acceptance criteria for ports in the vcpkg registry.
 
Given that this is a new library with no pre-existing user base and it’s not part of a well stablished suite of libraries, we believe that it is a good candidate for [self-hosting in a custom registry](https://learn.microsoft.com/produce/publish-to-a-git-registry). Users will still be able to install your library with vcpkg using either the [registries](https://learn.microsoft.com/vcpkg/concepts/registries) or [overlay ports](https://learn.microsoft.com/vcpkg/concepts/overlay-ports) features; and you get full control of when you update the port and the changes it includes.
 
Would you be interested in setting up a custom registry for your library? We can aid setting up your registry and providing instructions for your users to install your vcpkg port.

We are also interested in receiving feedback regarding the suggested alternative if you decide to give it a try.

I'm reopening this PR to serve as a way of communication with you regarding the suggested alternative of self-hosting this port. Feel free to use this thread to request assistance with said task.

jimwang118 commented:
@Kered13 Thanks for your PR, Is work still being done for this PR?

vicroms commented:
/azp run

Kered13 commented:
Sorry for the long radio silence. I had moved on to other projects for awhile. I have returned to this project and updated it, and I'm prepared to try submitting this again.

Kered13 committed:
Update versions/n-/neoslippi.json

Co-authored-by: Kai Pastor <dg0yt@darc.de>
